File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.aspectj.weaver.ast.Var;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        List<TbBrand> brandList = brandService.findAll();
        return brandList;
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page,int size){
        PageResult pageResult = brandService.findPage(page, size);
        return pageResult;
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {
            brandService.save(brand);
            return new Result(true,"增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"增加失败："+e.getMessage());
        }
    }

    /**
     * 修改
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }


    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        TbBrand tbBrand = brandService.findOne(id);
        return tbBrand;
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand,int page,int size){
        PageResult pageResult = brandService.findPage(tbBrand,page, size);
        return pageResult;
    }

    @RequestMapping("/selectBrandList")
    public List<Map> selectBrandList (){
        return brandService.selectBrandList();
    }
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope, $http,$controller,brandService) {

$controller('baseController',{$scope:$scope});
    //读取列表数据绑定到表单中
    $scope.findAll = function () {
        brandService.findAll.success(
            function (response) {
                $scope.list = response;
            }
        );
    }

    //新增
    $scope.save =function(){
        var object = null;
        if($scope.entity.id!=null){
            object =brandService.update($scope.entity);
        }else{
            object =brandService.add($scope.entity);
        }
        object.success(
            function (response) {
                $scope.reloadList();
            }
        );
    }
    //查找一个
    $scope.findOne =function(id){
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        );
    }
    $scope.delete = function(ids){
        brandService.delete($scope.selectIds).success(
            function (response){
                $scope.reloadList();
            }
        )
    }
    $scope.searchBrand={};
    $scope.search=function(pages,rows){
        brandService.search(pages,rows,$scope.searchBrand).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    }

});